provisioning/device/src/polling_state_machine.ts

```diff
@@ -69,6 +62,12 @@ export class  PollingStateMachine extends EventEmitter {
             });
           },
           cancel: (err, result, response, callback) => this._fsm.transition('cancelling', err, result, response, callback),
+          /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_033: [ If `disconnect` is called while in the middle of a `registrationRequest` operation, the operation shall be cancelled and the transport shall be disconnected. ] */
+          disconnect: (callback) => {
+            this._fsm.handle('cancel', null, null, null, (err) => {
+              this._fsm.transition('disconnecting', callback);
```

Comment:
>this._fsm.transition('disconnecting', callback); [](start = 14, length = 48)

since cancel induces a transition, i'd rather you call this._fsm.handle(disconnect, callback) here rather than directly a transition.